### PR TITLE
npc spawns

### PR DESCRIPTION
**Pull request by jarryd229**
npc spawns

### CODE REVIEW

File: game/src/main/kotlin/world/gregs/voidps/world/map/goblin_village/Goblins.kts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import world.gregs.voidps.engine.entity.character.player.Player
+import world.gregs.voidps.engine.event.on
+import world.gregs.voidps.world.interact.dialogue.*
+import world.gregs.voidps.world.interact.dialogue.type.choice
+import world.gregs.voidps.world.interact.dialogue.type.npc
+import world.gregs.voidps.world.interact.dialogue.type.player
+import kotlin.random.Random
+
+
+fun redgoblins(id: String) = id == "goblin_staff_red" || id == "goblin_shield_battleaxe_red" || id == "goblin_shield_spear_red" || id == "goblin_mohawk_staff_red" || id == "goblin_battleaxe_red" || id == "goblin_ponytail_staff_red" || id == "goblin_shield_warhammer_red"
```
Comment (GregHib) [inline game/src/main/kotlin/world/gregs/voidps/world/map/goblin_village/Goblins.kts]:
```suggestion
fun redGoblins(id: String) = id.startsWith("goblin_") && id.endsWith("_red")
```

File: game/src/main/kotlin/world/gregs/voidps/world/map/goblin_village/Goblins.kts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+fun greengoblins(id: String) = id == "goblin_staff_green" || id == "goblin_mohawk_staff_green" || id == "goblin_shield_battleaxe_green" || id == "goblin_shield_warhammer_green" || id == "goblin_shield_spear_green" || id == "goblin_staff_2_green"
```
Comment (GregHib) [inline game/src/main/kotlin/world/gregs/voidps/world/map/goblin_village/Goblins.kts]:
```suggestion
fun greenGoblins(id: String) = id.startsWith("goblin_") && id.endsWith("_green")
```

File: game/src/main/kotlin/world/gregs/voidps/world/map/goblin_village/Goblins.kts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+on<NPCOption>({ operate && redgoblins(target.id) && option == "Talk-to" }) { player: Player ->
```
Comment (GregHib) [inline game/src/main/kotlin/world/gregs/voidps/world/map/goblin_village/Goblins.kts]:
```suggestion
on<NPCOption>({ operate && redGoblins(target.id) && option == "Talk-to" }) { player: Player ->
```

File: game/src/main/kotlin/world/gregs/voidps/world/map/goblin_village/Goblins.kts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                He wearing green armour! General Bentnoze tell us
+                wear red!
+             """, largeHead = true)
+            npc<Angry>(npcId = "goblin_staff_green","But General Wartface say we must wear green!", largeHead = true)
+        }
+    }
+}
+
+
+on<NPCOption>({ operate && greengoblins(target.id) && option == "Talk-to" }) { player: Player ->
```
Comment (GregHib) [inline game/src/main/kotlin/world/gregs/voidps/world/map/goblin_village/Goblins.kts]:
```suggestion
on<NPCOption>({ operate && greenGoblins(target.id) && option == "Talk-to" }) { player: Player ->
```

File: game/src/main/kotlin/world/gregs/voidps/world/map/lumbridge/LumbridgeFlag.kts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import world.gregs.voidps.engine.entity.character.forceChat
+import world.gregs.voidps.engine.entity.character.player.Player
+import world.gregs.voidps.engine.entity.character.setAnimation
+import world.gregs.voidps.engine.entity.obj.ObjectOption
+import world.gregs.voidps.engine.event.on
+import world.gregs.voidps.engine.queue.softQueue
+
+
+on<ObjectOption>({ operate && target.id == "lumbridge_flag" && option == "Raise" }) { player: Player ->
+    player.setAnimation("lumbridge_flag_raise")
```
Comment (GregHib) [inline game/src/main/kotlin/world/gregs/voidps/world/map/lumbridge/LumbridgeFlag.kts]: Make sure to have `arriveDelay()` at the start of Object interactions; it delays the player until they've finished moving/right beside the object before starting to interact.

**Review by GregHib — CHANGES_REQUESTED** [review]

File: game/src/main/kotlin/world/gregs/voidps/world/map/lumbridge/LumbridgeFlag.kts (hunk 5 above)
Comment (jarryd229) [reply game/src/main/kotlin/world/gregs/voidps/world/map/lumbridge/LumbridgeFlag.kts]: ok iv added it now

**Review by GregHib — APPROVED** [review]

**Pull request merged**